ADR/0014-let-pipelines-proceed.md

```diff
@@ -0,0 +1,53 @@
+# 14. Let Pipelines Proceed
+
+* Date Decided: 2022-05-??
+* Date Documented: 2023-01-31
+
+## Status
+
+Accepted
+
+Relates to [ADR 13. Stonesoup Test Stream - API contracts](0013-integration-service-api-contracts.html)
+
+## Context
+
+The user's build pipeline includes scanning and linting tasks that operate on the source code and
+the built image (SAST, antivirus, clair, etc..). The purpose of these tasks is to find problems in
+the user's source code or dependencies and alert the user so they can take action ([STONE-459]).
+
+One frustration we've heard from users in previous systems is that they don't want to be blocked in
+their development or testing by complications in the build system, by compliance concerns. We want
+to fix that by offering a system that permits as much progress in the lifecycle of the user's
+application (build, test, and pre-production deployment) but which also protects production from
+non-compliant builds via mechanisms in the [enterprise contract].
+
+A problem we face: in Tekton, a failing TaskRun causes the whole PipelineRun to fail. If the purpose
+of our linting and scanning tasks is to find problems - which usually looks like failure with
+a non-zero exit code - how do they do their job without constantly breaking the user's build,
+stopping them from running integration tests, and stopping them from deploying candidate builds to
+their lower [Environments]?
+
+## Decision
+
+All scanning and linting TaskRuns should *succeed* even if they find problems in the content they
+are evaluating.
```

Comment:
Are there any types of Tasks which should explicitly _not_ always succeed?

Reply:
Not clear to me at this time.

The git clone task should fail if it fails (because it can't provide output to the next stages of the pipelinerun). The buildah task should fail if it fails (because it can't produce the image... which is needed by subsequent stages of the pipelinerun?)